src/constructs/iam/policies/dynamodb.ts

```diff
@@ -0,0 +1,32 @@
+import type { GuStack } from "../../core";
+import type { GuPolicyProps } from "./base-policy";
+import { GuAllowPolicy } from "./base-policy";
+
+interface GuDynamoDBPolicyProps {
+  tableName: string;
+}
+
+interface GuDynamoDBPolicyPropsWithActions extends GuPolicyProps, GuDynamoDBPolicyProps {
+  actions: string[];
+}
+
+abstract class GuDynamoDBPolicy extends GuAllowPolicy {
+  protected constructor(scope: GuStack, id: string = "GuDynamoReadPolicy", props: GuDynamoDBPolicyPropsWithActions) {
+    super(scope, id, {
+      actions: props.actions.map((action) => `dynamodb:${action}`),
+      resources: [`arn:aws:dynamodb:${scope.region}:${scope.account}:table/${props.tableName}`],
+    });
+  }
+}
+
+export class GuDynamoDBReadPolicy extends GuDynamoDBPolicy {
+  constructor(scope: GuStack, id: string = "GuDynamoDBReadPolicy", props: GuDynamoDBPolicyProps) {
```

Comment:
I think this will fail the linter check once it's rebased as the id can only have a default if the props are optional given that the default won't ever be used.